clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1352,8 +1361,42 @@ public void handle(OffsetCommitResponse commitResponse, RequestFuture<Void> futu
             Set<String> unauthorizedTopics = new HashSet<>();
 
             for (OffsetCommitResponseData.OffsetCommitResponseTopic topic : commitResponse.data().topics()) {
+                String topicName = topic.name();
```

Comment:
Now that we can rely on the version, we should use it here and simplify all this logic.